crates/bevy_animation/src/blending.rs

```diff
@@ -0,0 +1,297 @@
+use std::mem::size_of;
+
+use bevy_asset::{Asset, Handle, HandleUntyped};
+use bevy_math::prelude::*;
+use bevy_render::color::Color;
+
+use crate::interpolate::{
+    utils::{Scale2, Scale3},
+    Lerp,
+};
+
+/// Mask size used to blend properties, each bit corresponds to a single property
+pub type Mask = u32;
+
+/// Number of animated properties a type can hold
+pub const MASK_LIMIT: usize = size_of::<Mask>() * 8;
+
+#[derive(Default, Debug)]
+pub struct AnimatorBlending {
+    bits: Vec<Mask>,
+    weight_slots: usize,
+    weights: Vec<f32>,
+}
+
+impl AnimatorBlending {
+    #[inline(always)]
+    pub fn begin_blending(&mut self, entities: usize, weight_slots: usize) -> AnimatorBlendGroup {
+        self.bits.clear();
+        self.bits.resize(entities, 0);
```

Comment:
nit: I think i would try to be clearer in my intensions by using `self.bits = vec![0; entities];`

Same as for weights below. I am sure you are trying to optimize and reuse memory but with changing sizes I feel like this may not be the case.

Reply:
`self.bits = vec![0; entities];` will always allocate a new vec, while resize wont if the sizes match

Reply:
Are you not forcing a mismatch in size by using `clear()` before you resize?

Reply:
Sorry is not about the size but the vec capacity